common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -361,6 +366,69 @@ public static long getLong(byte[] data, int index) {
         return PlatformDependent0.getLong(data, index);
     }
 
+    private static long getLongSafe(byte[] bytes, int offset) {
+        if (BIG_ENDIAN_NATIVE_ORDER) {
+            return (long) bytes[offset] << 56 |
+                    ((long) bytes[offset + 1] & 0xff) << 48 |
+                    ((long) bytes[offset + 2] & 0xff) << 40 |
+                    ((long) bytes[offset + 3] & 0xff) << 32 |
+                    ((long) bytes[offset + 4] & 0xff) << 24 |
+                    ((long) bytes[offset + 5] & 0xff) << 16 |
+                    ((long) bytes[offset + 6] & 0xff) <<  8 |
+                    (long) bytes[offset + 7] & 0xff;
+        }
+        return (long) bytes[offset] & 0xFF |
+                ((long) bytes[offset + 1] & 0xff) << 8 |
+                ((long) bytes[offset + 2] & 0xff) << 16 |
+                ((long) bytes[offset + 3] & 0xff) << 24 |
+                ((long) bytes[offset + 4] & 0xff) << 32 |
+                ((long) bytes[offset + 5] & 0xff) << 40 |
+                ((long) bytes[offset + 6] & 0xff) << 48 |
+                ((long) bytes[offset + 7] & 0xff) << 56;
+    }
+
+    private static long getLongFromBytesSafe(CharSequence bytes, int offset) {
+        return (long) bytes.charAt(offset) & 0xFF |
```

Comment:
https://github.com/netty/netty/pull/4436#discussion_r43830826 applies to all of the `get*FromBytesSafe` methods. So yes I need to think about endianess for all of them :)